src/main/java/mate/academy/webshop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,189 @@
+            statement.executeUpdate();
+            deleteOrderProducts(order.getId());
+            putOrderProducts(order);
+            return order;
+        } catch (SQLException e) {
+            throw new RuntimeException(String
+                    .format("Can't update for order id:%d", order.getId()), e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long orderId) {
+        String query = "DELETE FROM orders WHERE order_id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, orderId);
+
+            deleteOrderProducts(orderId);
+            return statement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new RuntimeException(String
+                    .format("Can't delete order with id:%d", orderId), e);
+        }
+    }
+
+    @Override
+    public List<Order> getUserOrders(User user) {
+        String query = "SELECT * FROM orders WHERE user_id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, user.getId());
+            return getAllOrdersFromResultSet(statement.executeQuery());
+        } catch (SQLException e) {
+            throw new RuntimeException(String
+                    .format("Can't get list of orders for user id:%d", user.getId()), e);
+        }
+    }
+
+    public void putOrderProducts(Order order) {
+        String query = "INSERT INTO orders_products "
+                + "(order_id, product_id) VALUES (?, ?)";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Product product : order.getProducts()) {
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setLong(1, order.getId());
+                statement.setLong(2, product.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(String
+                    .format("Can't put products for order id:%d", order.getId()), e);
+        }
+    }
+
+    public Optional<Order> getOrderFromResultSet(ResultSet resultSet)
+            throws SQLException {
+        if (resultSet.next()) {
+            Long userId = resultSet.getLong("user_id");
+            Long orderId = resultSet.getLong("order_id");
+            return Optional.of(new Order(orderId, getOrderProducts(orderId), userId));
+        }
+        return Optional.empty();
+    }
+
+    public List<Order> getAllOrdersFromResultSet(ResultSet resultSet)
+            throws SQLException {
+        List<Order> orderList = new ArrayList<>();
+
+        while (resultSet.next()) {
+            orderList.add(getOrderFromResultSet(resultSet)
+                    .orElseThrow(SQLException::new));
```

Comment:
може вже краще так? :
```suggestion
                    .orElseThrow(DataProcessingException::new));
```
або RuntimeException

Reply:
Пофіксив.